cursor_shared.c

```diff
@@ -708,6 +717,38 @@ PHP_METHOD(MongoCursorInterface, dead)
 }
 /* }}} */
 
+/* {{{ proto array MongoCursor::getReadPreference()
+   Returns an array describing the read preference for this cursor. Tag sets will be included if available. */
+PHP_METHOD(MongoCursorInterface, getReadPreference)
+{
+	mongo_cursor *cursor = (mongo_cursor*)zend_object_store_get_object(getThis() TSRMLS_CC);
+
+	array_init(return_value);
+	add_assoc_string(return_value, "type", mongo_read_preference_type_to_name(cursor->read_pref.type), 1);
+	php_mongo_add_tagsets(return_value, &cursor->read_pref);
+}
+/* }}} */
+
+/* {{{ proto bool MongoCursor::setReadPreference(string read_preference [, array tags ])
```

Comment:
protos are wrong now, they miss "Interface". Variable names also should have a $ in front of them (I think)